source/elements/oneDAL/source/algorithms/clustering/kmeans.rst

```diff
@@ -1,153 +1,494 @@
+.. highlight:: cpp
+.. default-domain:: cpp
+
 =======
 K-Means
 =======
 
-K-Means is a clustering algorithm that partitions :math:`n` observations into
-:math:`k` clusters minimizing some criterion within each cluster. Each cluster
-is characterized by a representative point, called *a centroid*, and a cluster
-radius.
+The K-Means algorithm partitions :math:`n` samples into :math:`k` *clusters*
+minimizing some criterion. Each cluster is characterized by a representative
+point, called *a centroid*.
+
+Given the training set :math:`X = \{ x_1, \ldots, x_n \}` of
+:math:`p`-dimensional samples and a positive integer :math:`k`, the problem is
+to find a set :math:`C = \{ c_1, \ldots, c_k \}` of :math:`p`-dimensional
+centroids that minimize the objective function
+
+.. math::
+   \Phi_{X}(C) = \sum_{i = 1}^n d^2(x_i, C),
+
+where :math:`d^2(x_i, C)` is the squared Euclidean distance from :math:`x_i`
+to the closest center in :math:`C`,
+
+.. math::
+   d^2(x_i, C) = \min_{1 \leq j \leq k} \| x_i - c_j \|^2, \quad 1 \leq i \leq k.
+
+Expression :math:`\|\cdot\|` denotes :math:`L_2` norm.
+
+.. note::
+   In the general case, :math:`d` may be an arbitrary distance function. Current
+   version of the oneDAL spec defines only Euclidean distance case.
+
 
-Given the set :math:`X = \{ x_1, \ldots, x_n \}` of :math:`p`-dimensional
-feature vectors and a positive integer :math:`k`, the problem is to find a set
-:math:`C = \{ c_1, \ldots, c_k \}` of :math:`p`-dimensional vectors that
-minimize the objective function
+--------------
+Lloyd's method
+--------------
+
+The Lloyd's method [Lloyd82]_ consists in iterative updates of centroids by
+applying the alternating *Assignment* and *Update* steps, where :math:`t`
+denotes a number of the current iteration, e.g., :math:`C^{(t)} = \{ c_1^{(t)},
```

Comment:
"the index" instead of "a number" seems more accurate

and "e.g." seems irrelevant here, you can put ":" there instead of it, or split the sentence in two